execute/plugin_functions.go

```diff
@@ -204,11 +243,14 @@ func filterOutExecutedMessages(
 
 // truncateObservation truncates the observation to fit within the given maxSize after encoding.
 // It removes data from the observation in the following order:
-// For each chain, remove commit reports one by one, if the encoded observation is still too large,
-// remove the entire chain.
-// Keep repeating this process until the encoded observation fits within the maxSize or there's only
-// one chain with one report left
-// Note: This function doesn't split one report into multiple parts.
+// For each chain, pick last report and start *pseudoDelete message by message.
```

Comment:
nit: now that there's no bool map, I don't think there needs to be a term for this. Just say messages are removed one by one until the report is small enough.
```suggestion
// For each chain, pick last report and start removing messages one at a time.
```